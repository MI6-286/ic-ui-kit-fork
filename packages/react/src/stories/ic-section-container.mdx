import {
  Meta,
  Story,
  Canvas,
  Controls,
  Description,
} from "@storybook/blocks";
import { IcSectionContainer } from "../components";
import readme from "../../../web-components/src/components/ic-section-container/readme.md";
import * as IcSectionContainerStories from './ic-section-container.stories';

<Meta of={IcSectionContainerStories} />

<Description markdown={readme} />

export const defaultArgs = {
  aligned: null,
  fullHeight: false
}

### Left Aligned

<Canvas>
  <Story of={IcSectionContainerStories.LeftAligned} />
</Canvas>

### Center alignment

<Canvas>
  <Story of={IcSectionContainerStories.CenterAlignment} />
</Canvas>

### Full-width alignment

<Canvas>
  <Story of={IcSectionContainerStories.FullWidthAlignment} />
</Canvas>

### Full height

<Canvas>
  <Story of={IcSectionContainerStories.FullHeight} />
</Canvas>

### Playground

<Canvas>
  <Story of={IcSectionContainerStories.Playground} />
</Canvas>
