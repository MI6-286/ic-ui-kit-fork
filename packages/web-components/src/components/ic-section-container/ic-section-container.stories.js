import { html } from "lit-html";

export default {
  title: "Web Components/Section container",
  component: "ic-section-container",
};

export const LeftAligned = {
  render: () =>
    html`<ic-section-container>
      <main>
        <div style="display: flex; justify-content: space-between">
          <button>Start</button>
          <button>End</button>
        </div>
      </main>
    </ic-section-container>`,

  name: "Left aligned",
};

export const CenterAlignment = {
  render: () =>
    html`<ic-section-container aligned="center">
      <main>
        <div style="display: flex; justify-content: space-between">
          <button>Start</button>
          <button>End</button>
        </div>
      </main>
    </ic-section-container>`,

  name: "Center alignment",
};

export const FullWidthAlignment = {
  render: () =>
    html`<ic-section-container aligned="full-width">
      <main>
        <div style="display: flex; justify-content: space-between">
          <button>Start</button>
          <button>End</button>
        </div>
      </main>
    </ic-section-container>`,

  name: "Full-width alignment",
};

export const FullHeight = {
  render: () =>
    html`<ic-section-container aligned="full-width" full-height>
      <main>
        <div style="display: flex; justify-content: space-between">
          <button>Start</button>
          <button>End</button>
        </div>
      </main>
    </ic-section-container>`,

  name: "Full-height",
};
