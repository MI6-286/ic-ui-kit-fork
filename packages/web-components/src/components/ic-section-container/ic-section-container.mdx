import { Meta, Story, Canvas, Controls, Description } from "@storybook/blocks";
import { html } from "lit-html";
import readme from "./readme.md";
import * as IcSectionContainerStories from "./ic-section-container.stories";

<Meta of={IcSectionContainerStories} />

<Description markdown={readme} />

### Left aligned

<Canvas>
  <Story of={IcSectionContainerStories.LeftAligned} />
</Canvas>

### Center alignment

<Canvas>
  <Story of={IcSectionContainerStories.CenterAlignment} />
</Canvas>

### Full-width alignment

<Canvas>
  <Story of={IcSectionContainerStories.FullWidthAlignment} />
</Canvas>

### Full height

<Canvas>
  <Story of={IcSectionContainerStories.FullHeight} />
</Canvas>
