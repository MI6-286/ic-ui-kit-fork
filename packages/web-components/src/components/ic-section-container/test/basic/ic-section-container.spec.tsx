import { SectionContainer } from "../../ic-section-container";
import { newSpecPage } from "@stencil/core/testing";

describe("ic-section-container", () => {
  it("should render with left-aligned by default", async () => {
    const page = await newSpecPage({
      components: [SectionContainer],
      html: `<ic-section-container>
      <p>Test container</p>
      </ic-section-container>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render full-width", async () => {
    const page = await newSpecPage({
      components: [SectionContainer],
      html: `<ic-section-container aligned="full-width">
      <p>Test container</p>
      </ic-section-container>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render centered", async () => {
    const page = await newSpecPage({
      components: [SectionContainer],
      html: `<ic-section-container aligned="center">
      <p>Test container</p>
      </ic-section-container>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should remove vertical padding when full-height", async () => {
    const page = await newSpecPage({
      components: [SectionContainer],
      html: `<ic-section-container full-height>
      <p>Test container</p>
      </ic-section-container>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
