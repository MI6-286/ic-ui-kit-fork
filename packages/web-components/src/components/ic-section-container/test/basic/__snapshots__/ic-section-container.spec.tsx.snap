// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-section-container should remove vertical padding when full-height 1`] = `
<ic-section-container class="aligned-left no-vertical-padding" full-height="">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <p>
    Test container
  </p>
</ic-section-container>
`;

exports[`ic-section-container should render centered 1`] = `
<ic-section-container aligned="center" class="aligned-center">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <p>
    Test container
  </p>
</ic-section-container>
`;

exports[`ic-section-container should render full-width 1`] = `
<ic-section-container aligned="full-width" class="aligned-full-width">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <p>
    Test container
  </p>
</ic-section-container>
`;

exports[`ic-section-container should render with left-aligned by default 1`] = `
<ic-section-container class="aligned-left">
  <template shadowrootmode="open">
    <slot></slot>
  </template>
  <p>
    Test container
  </p>
</ic-section-container>
`;
